server/project.go

```diff
@@ -122,6 +122,37 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 		return nil
 	})
 
+	// Create a dedicated endpoint for fetching all projects.
+	// This is only used by Owner and DBA and we can thus enforce ACL on casbin.
+	g.GET("/project/all", func(c echo.Context) error {
```

Comment:
I'm thinking if we need a separate admin page? Shall we change existing "/project" endpoint to return all the projects if the user role is OWNER or DBA?

Reply:
I am inclined to separate the admin endpoint from the non-admin endpoint. I feel that's more readable and we can better enforce the ACL policy. wdyt?

Reply:
From an API perspective, it should be the same endpoint but with the right IAM. For example, cloud providers don't have different API endpoints for different project or organization roles.

Reply:
There's a tricky part of implementing in the same endpoint: the regular `Project` dashboard page would have both meanings:
1. you are a member of the projects listed on this page
2. you are the owner/administrator of the projects listed on this page

IMHO, we should at least make this straight if we are implementing in the same `/project` endpoint.

Reply:
> From an API perspective, it should be the same endpoint but with the right IAM. For example, cloud providers don't have different API endpoints for different project or organization roles. Also we can save some implementation on the frontend.

Separating the API is actually easier and makes the code more readable. Please check the frontend code change.

Mixing the endpoint has the complication as @dragonly mentioned. It makes code less readable, because there is quite a few if-else logic.

Also we may implement "Private project"  feature later, and that will add extra complexity if we share the code.

The way I view this is they are inherently separate endpoints, one is for non-admin API, one is for Admin API. The difference is more about the API itself (non-admin vs Admin), different role is a by-product. This is different from a normal API where different roles may have different access.

Another way to make this more explicit is to call it /project/admin or /admin/project to make it more explicit. The latter requires introducing an additional module, and will split the project-related implementation. So I would choose /project/admin if that sends a better signal.

Reply:
Or shall we take a flag "--admin" for the API which is more like "sudo xxx"? I think we need a good balance of # of APIs v.s. if-else.

Per, I think it's totally okay to have one central place to see all the resources instead of two different console pages. This is similar to Google cloud organization administrators should have the permission to see and manage all the resources in the organization using the same set of UIs.
---
1. you are a member of the projects listed on this page
2. you are the owner/administrator of the projects listed on this page
---

Reply:
>> Per, I think it's totally okay to have one central place to see all the resources instead of two different console pages. This is similar to Google cloud organization administrators should have the permission to see and manage all the resources in the organization using the same set of UIs.

I would say that's different, we are more like Google Cloud Admin which will have a project panel to manage All Google cloud projects. A closer comparison would be GitLab admin

Reply:
Changed per offline discussion.